fastparquet/core.py

```diff
@@ -246,3 +249,38 @@ def read_col(column, schema_helper, infile, use_cat=False,
         if se.converted_type is not None:
             out = convert(out, se)
     return out
+
+
+def read_row_group_file(fn, columns, *args, open=open):
+    with open(fn, mode='rb') as f:
+        return read_row_group(f, columns, *args)
+
+
+def read_row_group(file, rg, columns, categories, schema_helper, cats):
+    """
+    Access row-group in a file and read some columns into a data-frame.
+    """
+    out = {}
+
+    for column in rg.columns:
+        name = ".".join(column.meta_data.path_in_schema)
+        if name not in columns:
+            continue
+
+        use = name in categories if categories is not None else False
+        s = read_col(column, schema_helper, file, use_cat=use)
+        out[name] = s
+    out = pd.DataFrame(out, columns=columns)
+
+    # apply categories
+    for cat in cats:
+        # *Hard assumption*: all chunks in a row group have the
+        # same partition (correct for spark/hive)
+        partitions = re.findall("([a-zA-Z_]+)=([^/]+)/",
+                                rg.columns[0].file_path)
+        val = [p[1] for p in partitions if p[0] == cat][0]
+        codes = np.empty(rg.num_rows, dtype=np.int16)
```

Comment:
Did I choose int16 here? I'm pretty sure that the Categorical constructor, below, converts to int64